### ISSUE TITLE
IMPROVED PREDICTIONS OF MHC-PEPTIDE BINDING USING PROTEIN LANGUAGE MODELS

### ISSUE THREAD

**Issue by ghuijo** [One-Day-One-Paper/One-Day-One-Paper#324]
*MHC Class I peptide binding prediction 논문*

* BERT only uses the encoder part of the transformer since the goal is generating word embeddings
* there are a number of pre-trained self-supervised BERT-like models applied to protein data in the form of unlabeled amino acid sequences which can be very useful for many protein task-specific problems using transfer learning
<image src="https://user-images.githubusercontent.com/67627471/201526629-528d9237-d4ae-43b3-8e55-2600857d77c5.png" width=70% height=70%>

* ESM1b can be regarded as a transformer-based bidirectional language model (bi-LM)
* --> whether a given peptide sequence (same as text A) binds to a given MHC sequence (same as text B)
* generate the sequence "[BOS], M1, ..., Mp, [SEP], P1, ..., Pq, [EOS]" with length p + q + 3 as the ESM1b input
* use the embedding of [BOS] to be the MHC-peptide sequence-pair embedding vector v
* passing v through a softmax classifier layer / output the probability of binding / use it to compute the loss and apply back-propagation
* Compared to embedding MHC and peptide separately, this compound input allows the transformer to use the attention mechanism to further extract the interactive information between the amino acids in the MHC and peptide, thus, helping the binding prediction.
* After each GAT layer, we update the embedding vector for the amino acids and the special tokens and more GAT layers follow
* Comparison of AUC between our model and NetMHCpan (V4.1)
  <image src="https://user-images.githubusercontent.com/67627471/201526940-b4857e4f-cb14-4327-8834-571a095de906.png" width=40% height=40%>
